From 89c14b62bd61fb754363092d4b57a8a09d08d4f0 Mon Sep 17 00:00:00 2001
From: Jamin Lin <jamin_lin@aspeedtech.com>
Date: Tue, 2 Jul 2024 10:18:03 +0800
Subject: [PATCH] Support Yocto build
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

To make Yocto build system set its CC, AR and CFLASG
, replace "=" with "?=" for CC, AR and CFLAGS
environment variables of Makefile

Add LDFLAGS environment variable to make build process find the 
libraries from the Yocto build system in recipe-sysroot-native/usr/lib
instead of /usr/lib.

Signed-off-by: Jamin Lin <jamin_lin@aspeedtech.com>
---
 Makefile | 13 +++++++------
 1 file changed, 7 insertions(+), 6 deletions(-)

diff --git a/Makefile b/Makefile
index 6c62948..d5001c2 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,7 @@
-AR = /usr/bin/ar
-CC = /usr/bin/gcc
-CFLAGS = -Wall -O3
+AR ?= /usr/bin/ar
+CC ?= /usr/bin/gcc
+CFLAGS ?= -Wall -O3
+LDFLAGS ?=
 
 all: hss_lib.a \
      hss_lib_thread.a \
@@ -32,13 +33,13 @@ hss_verify.a: hss_verify.o hss_verify_inc.o hss_common.o hss_thread_single.o \
 	$(AR) rcs $@ $^
 
 demo: demo.c hss_lib_thread.a
-	$(CC) $(CFLAGS) demo.c hss_lib_thread.a -lcrypto -lpthread -o demo
+	$(CC) $(CFLAGS) $(LDFLAGS) demo.c hss_lib_thread.a -lcrypto -lpthread -o demo
 
 test_1: test_1.c lm_ots_common.o lm_ots_sign.o lm_ots_verify.o  endian.o hash.o sha256.o hss_zeroize.o
-	$(CC) $(CFLAGS) -o test_1 test_1.c lm_ots_common.o lm_ots_sign.o lm_ots_verify.o  endian.o hash.o sha256.o hss_zeroize.o -lcrypto
+	$(CC) $(CFLAGS) $(LDFLAGS) -o test_1 test_1.c lm_ots_common.o lm_ots_sign.o lm_ots_verify.o  endian.o hash.o sha256.o hss_zeroize.o -lcrypto
 
 test_hss: test_hss.c test_hss.h test_testvector.c test_stat.c test_keygen.c test_load.c test_sign.c test_sign_inc.c test_verify.c test_verify_inc.c test_keyload.c test_reserve.c test_thread.c test_h25.c hss.h hss_lib_thread.a
-	$(CC) $(CFLAGS) test_hss.c test_testvector.c test_stat.c test_keygen.c test_sign.c test_sign_inc.c test_load.c test_verify.c test_verify_inc.c test_keyload.c test_reserve.c test_thread.c test_h25.c hss_lib_thread.a -lcrypto -lpthread -o test_hss
+	$(CC) $(CFLAGS) $(LDFLAGS) test_hss.c test_testvector.c test_stat.c test_keygen.c test_sign.c test_sign_inc.c test_load.c test_verify.c test_verify_inc.c test_keyload.c test_reserve.c test_thread.c test_h25.c hss_lib_thread.a -lcrypto -lpthread -o test_hss
 
 hss.o: hss.c hss.h common_defs.h hash.h endian.h hss_internal.h hss_aux.h hss_derive.h
 	$(CC) $(CFLAGS) -c hss.c -o $@
-- 
2.25.1

